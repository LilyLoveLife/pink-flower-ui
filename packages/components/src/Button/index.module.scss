
$pink: pink;
.border-pink {
    border-color: $pink;
}
.customBtn {
    border-color: $pink;
}