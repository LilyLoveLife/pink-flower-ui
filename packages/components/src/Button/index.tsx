/*
 * @Description: 
 * @version: 
 * @Author: 
 * @Date: 2023-04-02 22:54:11
 * @LastEditors: Lily lily.song@hrtps.com
 * @LastEditTime: 2023-05-30 21:16:28
 */
import React, {useState} from 'react'
import cs from 'classnames'
import styles from './index.module.scss'

// import 'uno.css'

// import  './index.css'
// interface IButtonProps {
//     className?: string
// }


import {Button} from 'antd'
import { ButtonProps, ButtonType } from 'antd/lib/button'

interface IProps extends Omit<ButtonProps, 'type'> {
    type?: 'login' | ButtonType
}

const PinkButton = (props: IProps, ref: any) => {
  // return <button className='color-pink-400'>提交</button>
      const { className: customClassName, type, ...otherProps }: any = props
     let defaultClassName = cs({
        [styles.customBtn]: true,
        // [styles['border-pink']]: true,
        // [styles.login]: type === 'login',
        'color-pink-400': true
    })
    const className = customClassName ? `${defaultClassName} ${customClassName}` : defaultClassName

  return <Button className={className} {...otherProps}></Button>
    // const { className: customClassName, type, ...otherProps }: any = props
    // const iconClasses = classNames(
    //     // iconPrefixCls,
    //     // {
    //     //   [`${iconPrefixCls}-${sizeCls}`]: sizeCls,
    //     // },
    //     className
    // )
    // const InternalButton = (props: any, ref: any) => {
    //     const {
    //         loading = false,
    //         type = 'default',
    //         size: customizeSize,
    //         disabled: customDisabled,
    //         className,
            
    //     } = props
    //     return <Button {...props} ref={ref}></Button>
    // }

    // let defaultClassName = cs({
    //     // [styles.customBtn]: true,
    //     // [styles['border-pink']]: true,
    //     // [styles.login]: type === 'login',
    //     'color-pink-400': true
    // })
    // const className = customClassName ? `${defaultClassName} ${customClassName}` : defaultClassName
      
    // return <Button  type={['login'].includes(type) ? 'primary' : type} className={className} {...otherProps} ></Button>
    // return <Button  type={['login'].includes(type) ? 'primary' : type} className={'pink'} >submit</Button>

}
Object.assign(PinkButton, Button)
// PinkButton.xxx = xxx
export default PinkButton
// Object.assign(PinkButton, Button)

// export default React.forwardRef(PinkButton)