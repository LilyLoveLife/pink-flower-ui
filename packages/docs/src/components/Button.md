#### Button

```jsx
import React from 'react';
import { Button } from 'pink-flower-ui';
export default () => {
  return (
    <>
      <Button>submit</Button>
    </>
  );
};
```

#### API:

| 成员         | 说明             | 类型                                                  | 默认值  | 备注 |
| :----------- | :--------------- | :---------------------------------------------------- | :------ | :--- |
| `type`       | 设置按钮类型     | primary \| ghost \| dashed \| link \| text \| default | default | -    |
| `classNames` | 语义化结构 class | Record<SemanticDOM, string> \| undefined              |         |
| `disabled`   | 设置按钮失效状态 | boolean                                               | false   | -    |
| `styles`     | 语义化结构 style | Record<SemanticDOM, CSSProperties>                    | -       |
| `onClick`    | 点击按钮时的回调 | (event: MouseEvent) => void                           | -       | -    |
