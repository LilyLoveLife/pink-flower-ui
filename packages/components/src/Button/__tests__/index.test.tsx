/*
 * @Description:
 * @version:
 * @Author:
 * @Date: 2023-06-27 23:15:28
 * @LastEditors: Please set LastEditors
 * @LastEditTime: 2023-06-28 02:23:04
 */
// import React from 'react'
// import { create, act } from 'react-test-renderer' // 虚拟dom
import { render, screen } from "@testing-library/react"; // dom
import Button from "../index";

// jest.mock("react", () => {
//   const originReact = jest.requireActual("react");
//   const mUseRef = jest.fn();
//   return {
//     ...originReact,
//     useRef: mUseRef,
//   };
// });
describe("<Button />", () => {
  test("should render default", () => {
    // render(<Button />);
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });
  test("should render text", () => {
    // render(<Button />);
    const { container } = render(<Button>提交</Button>);
    // expect(container).toMatchSnapshot();
  });
});
