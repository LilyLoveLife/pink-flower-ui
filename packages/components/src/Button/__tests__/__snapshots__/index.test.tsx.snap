// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render default 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-ed5zg0 ant-btn-default color-pink-400"
    placeholder="请输入"
    type="button"
  />
</div>
`;
